spring-batch/src/main/java/org/baeldung/conditionalflow/NumberInfoDecider.java

```diff
@@ -0,0 +1,19 @@
+package org.baeldung.conditionalflow;
+
+import org.springframework.batch.core.ExitStatus;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.core.job.flow.FlowExecutionStatus;
+import org.springframework.batch.core.job.flow.JobExecutionDecider;
+
+public class NumberInfoDecider implements JobExecutionDecider {
+
+    @Override
+    public FlowExecutionStatus decide(JobExecution jobExecution, StepExecution stepExecution) {
+        if(jobExecution.getExitStatus().equals("UNKNOWN")) {
```

Comment:
This line will always evaluate to false as we are comparing ExitCode with String?

Reply:
No. See ExitCode equals method (it uses toString)